### PR TITLE
[NeoML] Transfer CDnnBlob data in threads pools

### PR DESCRIPTION
**Pull request by favorart**

### CODE REVIEW

File: NeoML/include/NeoML/Dnn/DnnBlob.h [hunk 1]
```diff
@@ -136,6 +136,11 @@ class NEOML_API CDnnBlob : public IObject {
 	// Copies the contents from another blob
 	void CopyFrom(const CDnnBlob* other);
 
+	// Transfers CDnnBlob data from other thread owner to this thread.
+	// CDnnBlob can be transfered, if at least one pointer to it is still exist, and on other side
```
Comment (FedyuninV) [inline NeoML/include/NeoML/Dnn/DnnBlob.h]: You can't call method of an objects if there is no pointers to it. So this part of the comment is excessive. And mb the users will change this association without stopping the original thread. I think it's better to rewrite it into smth like:

By default memory underneath each blob is associated with the thread on which the allocation of the blob has occurred.
This method switches this association to the calling thread.

File: NeoML/test/src/DnnBlobTest.cpp [hunk 2]
```diff
@@ -16,28 +16,172 @@ limitations under the License.
...
+{
+    const TMathEngineType met = MathEngine().GetType();
+    switch( met ) {
+        case MET_Cpu:
+            break;
+        case MET_Cuda:
+            if( TTransferType::PoolToPool == type ) {
+                break;
+            }
+        case MET_Metal:
```
Comment (FedyuninV) [inline NeoML/test/src/DnnBlobTest.cpp:76]: Modern compilers generate warnings when there's no break nor comment before switch case (they suspect us in forgetting `break`s)

File: NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h [hunk 3]
```diff
@@ -1184,6 +1184,11 @@ class NEOMATHENGINE_API IMathEngine : public IDnnEngine {
 	virtual void SetReuseMemoryMode( bool enable ) = 0;
 	virtual CMemoryHandle HeapAlloc( size_t count ) = 0;
 	virtual void HeapFree( const CMemoryHandle& handle ) = 0;
+	// Transfers memory handle from other thread owner to this thread.
+	// Caution! Do not use this method directly, only through the method CDnnBlob::TransferDataToThisThread()
+	// CDnnBlob can be transfered, if at least one pointer to it is still exist, and on other side
```
Comment (FedyuninV) [inline NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h]: The first two lines are good. The next two have the same problem as above (but here the original description is cool so you can simply remove the thing about pointers and about threads being finished)

Comment (favorart) [reply NeoMathEngine/include/NeoMathEngine/NeoMathEngine.h]: fixed

File: NeoML/include/NeoML/Dnn/DnnBlob.h (hunk 1 above)
Comment (favorart) [reply NeoML/include/NeoML/Dnn/DnnBlob.h]: fixed

File: NeoML/test/src/DnnBlobTest.cpp (hunk 2 above)
Comment (favorart) [reply NeoML/test/src/DnnBlobTest.cpp:76]: fixed

**Review by FedyuninV — APPROVED** [review]

**Pull request merged**